src/ray/raylet/task_util.h

```diff
@@ -0,0 +1,77 @@
+#ifndef RAY_RAYLET_TASK_UTIL_H
+#define RAY_RAYLET_TASK_UTIL_H
+
+#include "ray/protobuf/common.pb.h"
+#include "ray/raylet/task_spec.h"
+
+namespace ray {
+
+namespace raylet {
+
+/// Helper function for building the common fields of a `TaskSpec` message.
+/// See `common.proto` for the documentation of the arguments.
+///
+/// \param[out] message The `TaskSpec` message to build.
+void BuildCommonTaskSpec(
+    rpc::TaskSpec *message, const Language &language,
+    const std::vector<std::string> &function_descriptor, const JobID &job_id,
+    const TaskID &parent_task_id, uint64_t parent_counter, uint64_t num_returns,
+    const std::unordered_map<std::string, double> &required_resources,
+    const std::unordered_map<std::string, double> &required_placement_resources) {
+  // Set common fields.
+  message->set_type(rpc::TaskType::NORMAL_TASK);
+  message->set_language(language);
+  for (const auto &fd : function_descriptor) {
+    message->add_function_descriptor(fd);
+  }
+  message->set_job_id(job_id.Binary());
+  message->set_task_id(GenerateTaskId(job_id, parent_task_id, parent_counter).Binary());
+  message->set_parent_task_id(parent_task_id.Binary());
+  message->set_parent_counter(parent_counter);
+  message->set_num_returns(num_returns);
+  message->mutable_required_resources()->insert(required_resources.begin(),
+                                                required_resources.end());
+  message->mutable_required_placement_resources()->insert(
+      required_placement_resources.begin(), required_placement_resources.end());
+}
+
+/// Helper function for building the `ActorCreactionSpec` in a `TaskSpec` message.
+/// See `common.proto` for the documentation of the arguments.
+///
+/// \param[out] message The `TaskSpec` message to build.
```

Comment:
I think some additional documentation is required for these methods so that people know that the `message` argument have been passed to `BuildCommonTaskSpec` already.

Also, please document the other arguments (same for the other functions).

Reply:
1) The order doesn't actually matter. It's okay to call this function, before `BuildCommonTaskSpec`. It's just more intuitive that we build the common part, before other parts.
2) Regarding the documentation of the args, I was thinking that we could simplify copy-paste the documentation from `common.proto`. But when we update `common.proto` in the future, it's very likely that we'll forget to update here accordingly. So I ended up putting `See common.proto for the documentation of the arguments.` here.  Do you think it's okay? Or do you have any suggestions about how to better deal with this issue?

Reply:
The comment about "see common.proto" seems fine.